cmd/peer-rest-client.go

```diff
@@ -0,0 +1,450 @@
+	if err != nil {
+		return
+	}
+	defer http.DrainBody(respBody)
+	err = gob.NewDecoder(respBody).Decode(&info)
+	return info, err
+}
+
+// StartProfiling - Issues profiling command on the peer node.
+func (client *peerRESTClient) StartProfiling(profiler string) error {
+	values := make(url.Values)
+	values.Set(peerRESTProfiler, profiler)
+	respBody, err := client.call(peerRESTMethodStartProfiling, values, nil, -1)
+	if err != nil {
+		return err
+	}
+	defer http.DrainBody(respBody)
+	return nil
+}
+
+// DownloadProfileData - download profiled data from a remote node.
+func (client *peerRESTClient) DownloadProfileData() (data []byte, err error) {
+	respBody, err := client.call(peerRESTMethodDownloadProfilingData, nil, nil, -1)
+	if err != nil {
+		return
+	}
+	defer http.DrainBody(respBody)
+	err = gob.NewDecoder(respBody).Decode(&data)
+	return data, err
+}
+
+// DeleteBucket - Delete notification and policies related to the bucket.
+func (client *peerRESTClient) DeleteBucket(bucket string) error {
+	values := make(url.Values)
+	values.Set(peerRESTBucket, bucket)
+	respBody, err := client.call(peerRESTMethodDeleteBucket, values, nil, -1)
+	if err != nil {
+		return err
+	}
+	defer http.DrainBody(respBody)
+	return nil
+}
+
+// ReloadFormat - reload format on the peer node.
+func (client *peerRESTClient) ReloadFormat(dryRun bool) error {
+	values := make(url.Values)
+	if dryRun {
+		values.Set(peerRESTDryRun, "true")
+	} else {
+		values.Set(peerRESTDryRun, "false")
+	}
+
+	respBody, err := client.call(peerRESTMethodReloadFormat, values, nil, -1)
+	if err != nil {
+		return err
+	}
+	defer http.DrainBody(respBody)
+	return nil
+}
+
+// ListenBucketNotification - send listent bucket notification to peer nodes.
+func (client *peerRESTClient) ListenBucketNotification(bucket string, eventNames []event.Name,
+	pattern string, targetID event.TargetID, addr xnet.Host) error {
+	args := listenBucketNotificationReq{
+		EventNames: eventNames,
+		Pattern:    pattern,
+		TargetID:   targetID,
+		Addr:       addr,
+	}
+
+	values := make(url.Values)
+	values.Set(peerRESTBucket, bucket)
+
+	var reader bytes.Buffer
+	err := gob.NewEncoder(&reader).Encode(args)
+	if err != nil {
+		return err
+	}
+
+	respBody, err := client.call(peerRESTMethodBucketNotificationListen, values, &reader, -1)
+	if err != nil {
+		return err
+	}
+
+	defer http.DrainBody(respBody)
+	return nil
+}
+
+// SendEvent - calls send event RPC.
+func (client *peerRESTClient) SendEvent(bucket string, targetID, remoteTargetID event.TargetID, eventData event.Event) error {
+	args := SendEventRequest{
+		TargetID: remoteTargetID,
+		Event:    eventData,
+	}
+
+	values := make(url.Values)
+	values.Set(peerRESTBucket, bucket)
+
+	var reader bytes.Buffer
+	err := gob.NewEncoder(&reader).Encode(args)
+	if err != nil {
+		return err
+	}
+	respBody, err := client.call(peerRESTMethodSendEvent, values, &reader, -1)
+	if err != nil {
+		return err
+	}
+
+	var sendEventREsp SendEventResp
```

Comment:
should be sendEventResp, not sendEventREsp

Reply:
fixed